cmd/zed/agents/zfs_retire.c

```diff
@@ -41,6 +41,9 @@
 #include <libzutil.h>
 #include <libzfs.h>
 #include <string.h>
+#ifdef __APPLE__
+#include <libgen.h>
+#endif
```

Comment:
What is the `libgen.h` header for?  Is this something we should really be including on all platforms?

Reply:
agents/zfs_retire.c:246:17: warning: implicit declaration of function 'basename'
      is invalid in C99 [-Wimplicit-function-declaration]
                    dev_name, basename(spare_name));

Reply:
Sure looks like we should be adding it everywhere and we're just getting away with it on Linux!  No need to the guard macro here!

Reply:
We could possibly just use `pthread_setname_np(X)` in the child thread, and on Linux it is a MACRO to `pthread_setname_tid(pthread_self(), X)`.  Assuming that is still valid, and works, on Linux to set the name in the thread itself, but it does feel like APPLE is what is the odd-one-out wrt to API?

But to be honest, I'm having problems finding **something/anything** in macOS that actually lets me see thread-names. If there is no way to see them, I might as well just make Linux's `pthread_setname_tid()` to nothing, and not set them in macOS. Anyone checked? No luck in `ps`, `spindump` `activity monitor` `sample`

Reply:
From the documentation I could find it sure looked to me like macOS was the only platform which handled this differently.  Sorry, I've no idea what might actually use this information on macOS.  But if it's not actually useful it's tempting not to do anything for this at all there.

Reply:
Also checked `lldb attach`, and `core` files.  The sources show pthread is setting __proc_info name field, but it really seems like there is nothing that will display it. For now, I think I'll roll the change back, and in future if we find the names would be useful, I'll bring it back.